framework/include/dirackernels/DiracKernel.h

```diff
@@ -216,6 +216,9 @@ class DiracKernel :
   /// Derivative of u_dot wrt u
   const VariableValue & _du_dot_du;
 
+  /// fully consider duplicate points
+  const bool _allow_duplicate_points;
```

Comment:
Which way does this go?  Basically, what does `allow` mean?  Does it mean that duplicates are ignored or that they're counted and then multiplied by the residuals?

How about: `_multiply_contribution_by_point_multiplicity` or something more descriptive?